tests/convert_utf32_to_latin1_tests.cpp

```diff
@@ -0,0 +1,65 @@
+#include "simdutf.h"
+
+#include <array>
+#include <iostream>
+
+#include <tests/helpers/transcode_test_base.h>
+#include <tests/helpers/random_int.h>
+#include <tests/helpers/test.h>
+#include <memory>
+#include <chrono>
+
+namespace {
+  std::array<size_t, 7> input_size{7, 16, 12, 64, 67, 128, 256};
+
+  using simdutf::tests::helpers::transcode_utf8_to_utf16_test_base;
+
+  constexpr size_t trials = 1000;
+}
+
+TEST(convert_latin1_only) {
+  size_t counter = 0;
+  auto generator = [&counter]() -> uint32_t {
+    return counter++ & 0xFF; 
+  };
+
+  auto procedure = [&implementation](const char32_t* utf32, size_t size, char* latin1) -> size_t {
+    return implementation.convert_utf32_to_latin1(utf32, size, latin1);
+  };
+  auto size_procedure = [](const char32_t*, size_t size) -> size_t {
+    return size;
+  };
+  for (size_t size: input_size) {
+    simdutf::tests::helpers::transcode_utf32_to_latin1_test_base test(generator, size);
+    ASSERT_TRUE(test(procedure));
+    ASSERT_TRUE(test.check_size(size_procedure));
+  }
+}
+
+TEST(convert_fails_if_input_too_large) {
+  for (size_t j = 0; j < trials; j++) { 
+    uint32_t seed = std::chrono::system_clock::now().time_since_epoch().count();
```

Comment:
Don't do that @Nick-Nuon : the tests should be deterministic as much as possible. The problem with depending on the time (`std::chrono::system_clock::now().time_since_epoch().count()`) is that you may get that the tests fail, then you pull them up and suddenly they pass. Or vice versa. It is very annoying. 

Randomized tests are fine in the context of fuzzers for example, or in a context where you are provided the necessary data to reproduce the tests when an error is encountered. But here, it could just be flat out annoying.

So, as a rule, unless you have explicit reasons not to, the tests should be deterministic: if it fails when you run it once, it should fail twice if you run it twice, and so forth.

Reply:
Gotcha.

Taking a quick break for lunch and then I'm gonna fix this.